diff --git a/node_modules/@candypoets/nipworker/dist/react-native.js b/node_modules/@candypoets/nipworker/dist/react-native.js
index db6141d..1421e27 100644
--- a/node_modules/@candypoets/nipworker/dist/react-native.js
+++ b/node_modules/@candypoets/nipworker/dist/react-native.js
@@ -250,9 +250,9 @@ class pe extends j {
         }
         this.handleNativeMessage(s);
       }
-    }, this.eventEmitter.addListener(k, this.eventListener), this.appStateSubscription = L.addEventListener("change", (t) => {
+    }, this.eventEmitter.addListener(k, this.eventListener), this.hadSubscriptionsBeforeBackground = false, this.appStateSubscription = L.addEventListener("change", (t) => {
       const i = this.appState === "active", s = this.appState === "background" || this.appState === "inactive", r = t === "background" || t === "inactive";
-      this.appState = t, i && r && this.cleanup(), s && t === "active" && this.nativeModule.wake();
+      this.appState = t, i && r && (this.hadSubscriptionsBeforeBackground = this.getSubscriptionCount() > 0, this.cleanup()), s && t === "active" && this.hadSubscriptionsBeforeBackground && (this.hadSubscriptionsBeforeBackground = false, this.nativeModule.wake());
     }), ae(this), Promise.resolve().then(() => this.restoreSession());
   }
   isDeinitialized() {
